kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/ProcessServiceBase.java

```diff
@@ -65,11 +65,9 @@ public ProcessServiceBase(ProcessService processService, DefinitionService defin
 
     public String startProcess(String containerId, String processId, String marshallingType) {
         containerId = context.getContainerId(containerId, ContainerLocatorProvider.get().getLocator());
-        // Check for presence of process id
-        ProcessDefinition procDef = definitionService.getProcessDefinition(containerId, processId);
-        if( procDef == null ) {
-            throw new IllegalStateException("Unable to find process '" + processId + "' in container " + containerId);
-        }
+        // check validity of deployment and process id
+        definitionService.getProcessDefinition(containerId, processId);
```

Comment:
just wondering, wouldn't it be enough to just handle IllegalStateException in the *Resource classes and return 404 when received such exception?